elastic/beats
# [AWS Fargate] Add memory hard limit from container metadata and remove usage percentage

## Proposed commit message
Add memory hard limit from container metadata and remove usage percentage.


## Checklist




## Author's Checklist



## How to test this PR locally



1. Follow the AWS Fargate [blogpost](https://www.elastic.co/blog/elastic-agent-monitor-ecs-aws-fargate-elastic-observability) to create to set up the integration and create the resources.

2. Be sure that your resource has cmd execution rights, for example add `"enable-execute-command"` argument while creating the resource via the AWS CLI:

```
aws ecs create-service \
    --cluster cluster-name \
    --task-definition task-definition-name \
    --enable-execute-command \
    --service-name service-name \
    --desired-count 1
```

3.  Run the shell via AWS CLI via:
```
aws ecs execute-command --cluster cluster-name \
    --task task-id \
    --container container-name \
    --interactive \
    --command "/bin/sh"
```

4. Retrieve ECS_CONTAINER_METADATA_URI_V4 via ```echo $ECS_CONTAINER_METADATA_URI_V4```.

5. Fetch task data via `curl`:
```
curl $ECS_CONTAINER_METADATA_URI_V4/task
curl $ECS_CONTAINER_METADATA_URI_V4/task/stats
```

6. Check data with the one fetched from the agent deployed in AWS (note that the agent needs to be built from the code in this PR).

## Related issues


- Closes #37120
